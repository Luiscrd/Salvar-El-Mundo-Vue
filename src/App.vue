<template>
  <Header />
  <router-view />
  <Footer />
</template>

<script>
import { defineAsyncComponent } from 'vue'

export default {
  name: 'App',
  components: {
    Header: defineAsyncComponent( () => import('@/modules/shared/components/Header.vue')),
    Footer: defineAsyncComponent( () => import('@/modules/shared/components/Footer.vue')),
  }
}
</script>

<style>
html, body{
  margin: 0;
  padding: 0;
}

body{
  font-family: Arial, Helvetica, sans-serif;
  position: relative;
}

#app{
  position: relative;
}
</style>
