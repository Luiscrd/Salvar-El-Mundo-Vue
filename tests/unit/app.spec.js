import { mount } from '@vue/test-utils'
import App from '@/App.vue'

const mockStore = {
    getters: jest.fn()
}

jest.mock('vuex', () => ({
    useStore: () => mockStore
}))

describe('App Component', () => {

    let wrapper;

    beforeEach(() => {

        wrapper = mount(App)

    });

    test('Debe de hacer mach con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

})