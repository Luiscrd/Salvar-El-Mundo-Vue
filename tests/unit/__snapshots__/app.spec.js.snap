// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Debe de hacer mach con el snapshot 1`] = `
<div class="toolbar" role="banner"><img width="100" alt="Lc Logo" src=""><span>Salvar al mundo - Vue</span><img width="50" alt="Vue Logo" src="">
  <div class="spacer"></div><a href="https://github.com/Luiscrd" target="_blank" class="link-github"><span class="span-lc">LcrPrograming</span><img src="" alt="Github" class="logo-github"></a>
</div>
<router-view></router-view>
<div class="footer"><span class="by-footer">By: LcrPrograming</span><a href="https://www.linkedin.com/in/luis-carballo-rua-3b4763248/" target="_blank"><img src="" alt="Linked In" class="logo-liked"></a></div>
`;
